src/resolvers/exotics/git-resolver.js

```diff
@@ -12,8 +12,10 @@ import Git from '../../util/git.js';
 
 const urlParse = require('url').parse;
 
+const GIT_PROTOCOL_PATTERN = /git\+.+:/;
```

Comment:
Since we're using patterns now, do you think we can remove `GIT_PROTOCOLS` and special casing for `https://` + ends with `.git` custom rules and bake them into the pattern in the future?